server.go

```diff
@@ -130,3 +132,16 @@ func (s *server) Shutdown() {
 		s.logger.Error("Failed to shutdown server", zap.Error(err))
 	}
 }
+
+func (s *server) clientAddress(req *http.Request) string {
+	rip, port, err := net.SplitHostPort(req.RemoteAddr)
+	if err != nil {
+		s.logger.Error("Error parsing RemoteAddr", zap.String("RemoteAddr", req.RemoteAddr))
+		return req.RemoteAddr
+	}
+	ip := req.Header.Get(s.conf.trustedHeader)
+	if ip == "" {
+		ip = rip
+	}
+	return net.JoinHostPort(ip, port)
+}
```

Comment:
Nice work! I think this is good to merge as is already but I wanted to follow up on this from your original issue (#3):

> Since there could be a series of reverse proxies, load balancers, etc. in front of the server, the trusted field could contain a comma and space list of IP addresses. We could report back the entire header value or (perhaps more usefully) simply the first.

Currently if the header is a list of addresses, will `net.JoinHostPort` still work? Should we have a unit test for this case as well?